agent/reverseproxy/reverseproxy.go

```diff
@@ -38,6 +39,13 @@ func NewReverseProxy(conf config.ListenerConfig, logger log.Logger) *ReverseProx
 		timeout: conf.Timeout,
 		logger:  logger,
 	}
+	if conf.InsecureSkipVerify {
+		proxy.Transport = &http.Transport{
```

Comment:
When `ReverseProxy.Transport` is `nil`, it uses `http.DefaultTransport`. Therefore if you set to a custom transport, your not just changing `TLSClientConfig`, but overriding all the fields from `http.DefaultTransport`?

Therefore I think this should be:
```go
transport := http.DefaultTransport
transport.TLSClientConfig = ...
proxy.Transport = transport
```

Also, could you move this before `rp := &ReverseProxy`

Reply:
I'm still quite new to Go, I had to work around a compilation error due to the Transport interface.